### PR TITLE
[sway/workspaces] feat: add high-priority-named optiion

### PR DESCRIPTION
**Pull request by Anakael**
Resolves: comments in https://github.com/Alexays/Waybar/issues/2188

### CODE REVIEW

File: src/modules/sway/workspaces.cpp [hunk 1]
```diff
@@ -279,9 +284,24 @@ Gtk::Button &Workspaces::addButton(const Json::Value &node) {
 }
 
 std::string Workspaces::getIcon(const std::string &name, const Json::Value &node) {
-  std::vector<std::string> keys = {"urgent", "focused", name, "visible", "default"};
```
Comment (Alexays) [inline src/modules/sway/workspaces.cpp:282]: Normal to remove visible?

File: src/modules/sway/workspaces.cpp [hunk 2]
```diff
@@ -279,9 +284,24 @@ Gtk::Button &Workspaces::addButton(const Json::Value &node) {
...
+
+      it = std::find_if(high_priority_named_.begin(), high_priority_named_.end(),
+                        [&](const std::string &member) {
+                          return trimWorkspaceName(member) == trimWorkspaceName(name);
+                        });
+      if (it != high_priority_named_.end()) {
+        return config_["format-icons"][trimWorkspaceName(name)].asString();
+      }
+    }
+    if (key == "focused" || key == "urgent") {
```
Comment (Alexays) [inline src/modules/sway/workspaces.cpp:304]: Same here

**Comment by Anakael** [comment]
There is no documentation about that key neather in .scd or in wiki. So it's literally undocumented option.And it seems to overlap with `default` key.  21.07.2023, 10:41, "Alexis Rouillard" ***@***.***>:
@Alexays commented on this pull request.



In src/modules/sway/workspaces.cpp:
> @@ -279,9 +284,24 @@ Gtk::Button &Workspaces::addButton(const Json::Value &node) {
 }
 
 std::string Workspaces::getIcon(const std::string &name, const Json::Value &node) {
-  std::vector<std::string> keys = {"urgent", "focused", name, "visible", "default"};

Normal to remove visible?

File: src/modules/sway/workspaces.cpp (hunk 1 above)
Comment (Anakael) [reply src/modules/sway/workspaces.cpp:282]: There is no documentation about that key neather in .scd or wiki. So it's literally undocumented option.
And it seems to overlap with `default` key.

**Comment by Alexays** [comment]
Thanks!

**Pull request merged**